redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,685 @@
+        std::variant<std::string>(value));
+}
+
+inline void
+    getLicenseActivationAck(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& status,
+                            const std::string& licenseString)
+{
+    if (status == "com.ibm.License.LicenseManager.Status.ActivationFailed")
+    {
+        BMCWEB_LOG_WARNING("LicenseActivationStatus: ActivationFailed");
+        messages::installFailed(asyncResp->res, "ActivationFailed");
+    }
+    else if (status == "com.ibm.License.LicenseManager.Status.InvalidLicense")
+    {
+        BMCWEB_LOG_WARNING("LicenseActivationStatus: InvalidLicense");
+        messages::invalidLicense(asyncResp->res);
+    }
+    else if (status == "com.ibm.License.LicenseManager.Status.IncorrectSystem")
+    {
+        BMCWEB_LOG_WARNING("LicenseActivationStatus: IncorrectSystem");
+        messages::notApplicableToTarget(asyncResp->res);
+    }
+    else if (status ==
+             "com.ibm.License.LicenseManager.Status.IncorrectSequence")
+    {
+        BMCWEB_LOG_WARNING("LicenseActivationStatus: IncorrectSequence");
+        messages::notApplicableToTarget(asyncResp->res);
+    }
+    else if (status == "com.ibm.License.LicenseManager.Status.InvalidHostState")
+    {
+        BMCWEB_LOG_WARNING("LicenseActivationStatus: InvalidHostState");
+        messages::installFailed(asyncResp->res, "InvalidHostState");
+    }
+    else if (status == "com.ibm.License.LicenseManager.Status.Activated")
+    {
+        BMCWEB_LOG_INFO("License Activated");
+        messages::licenseInstalled(asyncResp->res, licenseString);
+    }
+    else
+    {
+        auto ires = asyncResp->res.resultInt();
+        BMCWEB_LOG_INFO("License Activation Failed: {}", ires);
+        messages::internalError(asyncResp->res);
+    }
+    // Reset LicenseString D-bus property to empty string
+    // after request status populated by pldmd.
+    resetLicenseString(asyncResp);
+}
+static void
+    handleLicensePost(const crow::Request& req,
+                      const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::string licenseString;
+    if (!json_util::readJsonAction(req, asyncResp->res, "LicenseString",
+                                   licenseString))
+    {
+        return;
+    }
+
+    if (licenseString.empty())
+    {
+        messages::propertyMissing(asyncResp->res, "LicenseString");
+    }
+
+    // Only allow one License install at a time
+    if (getLicenseActivationStatusMatch() != nullptr)
+    {
+        messages::resourceInUse(asyncResp->res);
+        return;
+    }
+
+    std::shared_ptr<boost::asio::steady_timer> timeout =
+        std::make_shared<boost::asio::steady_timer>(
+            crow::connections::systemBus->get_io_context());
+    timeout->expires_after(std::chrono::seconds(20));
+    crow::connections::systemBus->async_method_call(
+        [timeout, asyncResp,
+         licenseString](const boost::system::error_code ec) {
```

Comment:
```
const boost::system::error_code& ec
```

Reply:
I am not sure how addMessageToErrorJson this can be avoided  with licenseInstalled ?